nix/sources.json

```diff
@@ -1,50 +1,62 @@
-    },
-    "ipfs-key": {
-        "branch": "master",
-        "description": "A program to generate ipfs private keys",
-        "homepage": null,
-        "owner": "fission-suite",
-        "repo": "ipfs-key",
-        "rev": "9171388a3ecfc41966e13a9a56745b3b2eda0463",
-        "sha256": "195h3n38b9cnq3r5fip2nwkqwnhws45s8rc17j92sa3snf8nxygc",
-        "type": "tarball",
-        "url": "https://github.com/fission-suite/ipfs-key/archive/9171388a3ecfc41966e13a9a56745b3b2eda0463.tar.gz",
-        "url_template": "https://github.com/<owner>/<repo>/archive/<rev>.tar.gz"
-    },
-    "niv": {
-        "branch": "master",
-        "description": "Easy dependency management for Nix projects",
-        "homepage": "https://github.com/nmattia/niv",
-        "owner": "nmattia",
-        "repo": "niv",
-        "rev": "9d35b9e4837ab88517210b1701127612c260eccf",
-        "sha256": "0q50xhnm8g2yfyakrh0nly4swyygxpi0a8cb9gp65wcakcgvzvdh",
-        "type": "tarball",
-        "url": "https://github.com/nmattia/niv/archive/9d35b9e4837ab88517210b1701127612c260eccf.tar.gz",
-        "url_template": "https://github.com/<owner>/<repo>/archive/<rev>.tar.gz"
-    },
-    "nixpkgs": {
-        "branch": "nixpkgs-unstable",
-        "description": "A read-only mirror of NixOS/nixpkgs tracking the released channels. Send issues and PRs to",
-        "homepage": "https://github.com/NixOS/nixpkgs",
-        "owner": "NixOS",
-        "repo": "nixpkgs",
-        "rev": "cfed29bfcb28259376713005d176a6f82951014a",
-        "sha256": "034m892hxygminkj326y7l3bp4xhx0v154jcmla7wdfqd23dk5xm",
-        "type": "tarball",
-        "url": "https://github.com/NixOS/nixpkgs/archive/cfed29bfcb28259376713005d176a6f82951014a.tar.gz",
-        "url_template": "https://github.com/<owner>/<repo>/archive/<rev>.tar.gz"
-    }
+  "ipfs": {
+      "branch": "with-s3",
+      "description": "IPFS implementation in Go",
+      "homepage": "https://ipfs.io",
+      "owner": "fission-suite",
+      "repo": "go-ipfs",
+      "rev": "72c52d0ea145d43124be72c5f1367592320dc0ea",
+      "sha256": "047fr3ahpw8lnj1mjkrkhf1b2vw9687yks45bwrfqb0kaan0045k",
+      "type": "tarball",
+      "url": "https://github.com/fission-suite/go-ipfs/archive/72c52d0ea145d43124be72c5f1367592320dc0ea.tar.gz",
+      "url_template": "https://github.com/<owner>/<repo>/archive/<rev>.tar.gz"
+  },
+  "ipfs-key": {
+      "branch": "master",
+      "description": "A program to generate ipfs private keys",
+      "homepage": null,
+      "owner": "fission-suite",
+      "repo": "ipfs-key",
+      "rev": "9171388a3ecfc41966e13a9a56745b3b2eda0463",
+      "sha256": "195h3n38b9cnq3r5fip2nwkqwnhws45s8rc17j92sa3snf8nxygc",
+      "type": "tarball",
+      "url": "https://github.com/fission-suite/ipfs-key/archive/9171388a3ecfc41966e13a9a56745b3b2eda0463.tar.gz",
+      "url_template": "https://github.com/<owner>/<repo>/archive/<rev>.tar.gz"
+  },
+  "ipfs-migrator": {
+      "branch": "with-s3-v1",
```

Comment:
Just trying to understand how the go-ipfs bits work here. I'm guessing that this line is the main change?

Reply:
yeah you'll notice it's pointing to our fork of the migration tool: https://github.com/fission-suite/fs-repo-migrations

Reply:
which as the s3 plugin built directly in: https://github.com/fission-suite/fs-repo-migrations/commit/8fe6cf6de0861027d36856c4d6e69a885c5aafc2#diff-5aeeba508413ab7886c0ecf2a0e62a783360c3405cb39c24fd8948069b7d2685R143

Reply:
`v1` referring to the fact that we're on v1.7.1 of the migration tool instead of v2.0.1 which has a different api & multiple binaries